### PR TITLE
Implemented words and characters counter and added it in example

### PR DESCRIPTION
**Pull request by sureshsevarthi**

### CODE REVIEW

File: example/src/Counter.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * Words and characters counter example for Material Slate
+ */
+export default function Counter() {
+  // Holds the value of the editor
+  const [value, setValue] = useState(initialValue())
+
+  // An instance of material editor. It is an slate editor with a few more functions
+  const editor = useMemo(() => createMaterialEditor(), [])
+  // words length
+  const wordsLength = editor.getWordsLength(value)
```
[the remark this answers is not in the document]
Comment (merlos) [reply example/src/Counter.js:26]: In any case, these should not be required if wordLength and charLength are not required as props. Please, try that first.

File: src/components/Counters/WordCounter.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ *
+ * WordCounter for editor
+ * It displays the number words, below the editor
+ *  - If maxWords = 200, wordsLength = 90 `Ex: 90/200 words` will display in the counter
+ *  - When maxWords is undefined, wordsLength = 90 `Ex: 90 words` will be displayed
+ * Word counter will be displayed with error color, when wordLength exceeds maxWords
+ */
+export default function WordCounter({ wordsLength, maxWords }) {
```
[the remark this answers is not in the document]
Comment (sureshsevarthi) [reply src/components/Counters/WordCounter.jsx:24]: Initially i used useEffects. In order to get useEditor and useEffects, we have to pass wordCounter inside the MaterialSlate as children. So word counter is displaying inside the Material Slate border.

File: src/slate/createMaterialEditor.js [hunk 3]
```diff
@@ -6,35 +6,28 @@ import { withHistory } from 'slate-history'
...
-              withReact(
-                createEditor()
-              )
-            )
-          )
-        )
-      )
-  editor.editorId=editorId
+export default function createMaterialEditor(editorId = 'default') {
+  const editor = withCounter(
```
[the remark this answers is not in the document]
Comment (sureshsevarthi) [reply src/slate/createMaterialEditor.js:28]: Ok I will make it as optional

File: src/components/Counters/WordCounter.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (merlos) [reply src/components/Counters/WordCounter.jsx:24]: Counters must be inside MaterialSlate.
Try adding a divider before the counters in the example.

File: example/src/Counter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sureshsevarthi) [reply example/src/Counter.js]: Resolved the it by using useSlate

**Pull request merged**